packages/editor/src/store/reducer.js

```diff
@@ -142,18 +142,34 @@ export const editor = flow( [
 ] )( {
 	// Track whether changes exist, resetting at each post save. Relies on
 	// editor initialization firing post reset as an effect.
-	blocks: withChangeDetection( {
-		resetTypes: [ 'SETUP_EDITOR_STATE', 'REQUEST_POST_UPDATE_START' ],
-	} )( ( state = { value: [] }, action ) => {
-		switch ( action.type ) {
-			case 'RESET_EDITOR_BLOCKS':
-				if ( action.blocks === state.value ) {
-					return state;
-				}
-				return { value: action.blocks };
-		}
+	blocks: flow( [
+		combineReducers,
+		withChangeDetection( {
+			resetTypes: [ 'SETUP_EDITOR_STATE', 'REQUEST_POST_UPDATE_START' ],
+		} ),
+	] )( {
+		value( state = [], action ) {
```

Comment:
@epiqueras Since there is no `editor` recuser anymore, could you point me in the right direction where blocks are (and footnotes could be) stored with change detection and history?

Reply:
https://github.com/WordPress/gutenberg/blob/79d28866952ec3976eafad41d90458aa9d266455/packages/core-data/src/reducer.js#L154

https://github.com/WordPress/gutenberg/blob/79d28866952ec3976eafad41d90458aa9d266455/packages/core-data/src/reducer.js#L306